lib/actions/device.coffee

```diff
@@ -53,7 +53,8 @@ exports.list =
 		.tap (devices) ->
 			devices = _.map devices, (device) ->
 				device.dashboard_url = balena.models.device.getDashboardUrl(device.uuid)
-				device.application_name = device.belongs_to__application[0].app_name
+				device.application_name =
+					if device.belongs_to__application[0] then device.belongs_to__application[0].app_name else 'N/a'
```

Comment:
Not 100% sure, but I would think that `belongs_to__application` would be an empty array or even undefined in case the application was not accessible, so that `belongs_to__application[0]` might raise a TypeError:
```
$ node
Welcome to Node.js v12.13.0.
Type ".help" for more information.
> let a;
undefined
> a[0]
Thrown:
TypeError: Cannot read property '0' of undefined
```

Reply:
The CoffeeeScript [Existential Operator](https://coffeescript.org/#existential-operator) might come in handy in this case. Maybe (needs testing) the syntax would be:
```
if device.belongs_to__application?[0] then ...
```

Reply:
@pdcastro I believe in these cases `belongs_to__application` is sent as empty array from API.  E.g.:

```
{
  "d": [
	{
	  "belongs_to__application": [
	  ],
```

The original error that Shaun reported would suggest this:
```
TypeError: Cannot read property 'app_name' of undefined
```
Otherwise we'd expect the error you mentioned.

Still, probably good idea to check array too! (Will update)

Reply:
Updated with existential operator and tested for missing `belongs_to__application`.